### PR TITLE
Add radial search feature to main branch

### PR DESCRIPTION
**Pull request by junqiu-lei**
### Description
This PR:
* Updates `MINIMAL_SUPPORTED_VERSION_FOR_RADIAL_SEARCH` to 2.14
* Contains radial search PRs from feature branch: 
  * https://github.com/opensearch-project/k-NN/pull/1498
  * https://github.com/opensearch-project/k-NN/pull/1546
  * https://github.com/opensearch-project/k-NN/pull/1589
  * https://github.com/opensearch-project/k-NN/pull/1609. 

### Benchmark
https://github.com/opensearch-project/k-NN/issues/814#issuecomment-2060195145

### Issues Resolved
Part of https://github.com/opensearch-project/k-NN/issues/814
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ## [Unreleased 2.x](https://github.com/opensearch-project/k-NN/compare/2.13...2.x)
 ### Features
+* Support distance type radius search for Lucene engine [#1498](https://github.com/opensearch-project/k-NN/pull/1498)
```
Comment (jmazanec15) [inline CHANGELOG.md:17]: Can we consolidate into 1 change?

Comment (junqiu-lei) [reply CHANGELOG.md:17]: Sure

Comment (junqiu-lei) [reply CHANGELOG.md]: Updated

File: src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java [hunk 2]
```diff
@@ -128,6 +216,12 @@ public KNNQueryBuilder(StreamInput in) throws IOException {
             if (isClusterOnOrAfterMinRequiredVersion("ignore_unmapped")) {
                 ignoreUnmapped = in.readOptionalBoolean();
             }
+            if (isClusterOnOrAfterMinRequiredVersion(KNNConstants.RADIAL_SEARCH_KEY)) {
+                max_distance = in.readOptionalFloat();
+            }
+            if (isClusterOnOrAfterMinRequiredVersion(KNNConstants.RADIAL_SEARCH_KEY)) {
+                min_score = in.readOptionalFloat();
+            }
```
Comment (martin-gaievski) [inline src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java:224]: nit: we can have both reads under case if case. not a blocker, can be done in a refactoring PR in future

**Review by martin-gaievski — APPROVED** [review]

**Review by ryanbogan — APPROVED** [review]
LGTM!

**Comment by ryanbogan** [comment]
There are a few tests that are failing still, though

File: src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java (hunk 2 above)
Comment (junqiu-lei) [reply src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java:224]: Sure, we can do it in future PR

**Comment by junqiu-lei** [comment]
> There are a few tests that are failing still, though

The failing test isn’t related to the PR, I saw same failings in other recent PRs.

**Pull request merged**